pkg/limayaml/default.TEMPLATE.yaml

```diff
@@ -58,17 +58,21 @@ video:
   # Default: "none"
   display: "none"
 
-#UNIMPLEMENTED| provision:
-#UNIMPLEMENTED|   # `system` is executed with the root privilege
-#UNIMPLEMENTED|   system: |
-#UNIMPLEMENTED|     #!/bin/bash
-#UNIMPLEMENTED|     set -eux -o pipefail
-#UNIMPLEMENTED|     export DEBIAN_FRONTEND=noninteractive
-#UNIMPLEMENTED|     apt-get install -y vim
-#UNIMPLEMENTED|   # `user` is executed without the root privilege
-#UNIMPLEMENTED|   user: |
-#UNIMPLEMENTED|     #!/bin/bash
-#UNIMPLEMENTED|     set -eux -o pipefail
-#UNIMPLEMENTED|     cat <<EOF > ~/.vimrc
-#UNIMPLEMENTED|     set number
-#UNIMPLEMENTED|     EOF
+# provision:
+#   # `system` is executed with the root privilege
+#   system: |
+#     #!/bin/bash
+#     set -eux -o pipefail
+#     export DEBIAN_FRONTEND=noninteractive
+#     apt-get install -y vim
+#   # `user` is executed without the root privilege
+#   user: |
+#     #!/bin/bash
+#     set -eux -o pipefail
+#     cat <<EOF > ~/.vimrc
+#     set number
+#     EOF
+
+# Provision a bare VM without containerd and nerdctl
+# Default: false
+bare-vm: false
```

Comment:
This should be like :

```yaml
containerd:
  # Default: true
  enable: false
```

Reply:
or maybe:
```yaml
containerd:
  # Enable system-wide containerd (systemctl enable containerd)
  # Default: false
  system: false
  # Enable user-scoped containerd (systemctl --user enable containerd)
  # Default: true
  user: true
```

When `!systemd && !user`, containerd does not need to be installed at all.

Reply:
How can you have a "Default: true" for a boolean?

Should it be done like this:

```
        y.Containerd.Enable = true
	if err := yaml.Unmarshal(b, &y); err != nil {
		return nil, err
	}
	FillDefault(&y)
```

It is a little weird to split default handling into 2 places.

Reply:
Sorry, didn't see your second comment, but the question remains: how to implement a `true` default; it cannot be done after the call to `Unmarshal` because `false` is the null-value for booleans, so you don't know if it was a missing value, or if the user set it to `false`. Anyways, it can be done by pre-setting the field (I think; needs testing), but it splits default setting into 2 different places.

As for having both `containerd.system` and `containerd.user` settings, what should happen if both are set to `true`? Would it be better to have a single `containerd` setting that can be set to `system|user|none` instead?

Reply:
> How can you have a "Default: true" for a boolean?

Define the field as `*bool`, and let `FillDefault` set true when the value is nil.

Reply:
> As for having both containerd.system and containerd.user settings, what should happen if both are set to true? 

The both should be enabled. Run `systemctl enable --now` for system-wide containerd, and run `containerd-rootless-setuptool.sh install` for user (rootless) containerd.

> Would it be better to have a single containerd setting that can be set to system|user|none instead?

No, as it is 100% fine to have both system-wide containerd and user containerd.